Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/FileShareServiceCacheTest.cs

```diff
@@ -0,0 +1,219 @@
+using UKHO.MaritimeSafetyInformation.Common.Models.NoticesToMariners;
+using UKHO.MaritimeSafetyInformation.Web.Services;
+using UKHO.MaritimeSafetyInformation.Web.Services.Interfaces;
+
+namespace UKHO.MaritimeSafetyInformation.Web.UnitTests.Services
+{
+    [TestFixture]
+    public class FileShareServiceCacheTest
+    {
+        private IAzureTableStorageClient _azureTableStorageClient;
+        private IOptions<CacheConfiguration> _cacheConfiguration;
+        private IAzureStorageService _azureStorageService;
+        private IFileShareServiceCache _fileShareServiceCache;
+        private ILogger<FileShareServiceCache> _logger;
+        private IUserService _userService;
+
+        [SetUp]
+        public void Setup()
+        {
+            _azureTableStorageClient = A.Fake<IAzureTableStorageClient>();
+            _cacheConfiguration = A.Fake<IOptions<CacheConfiguration>>();
+            _logger = A.Fake<ILogger<FileShareServiceCache>>();
+            _fileShareServiceCache = A.Fake<IFileShareServiceCache>();
+            _azureStorageService = A.Fake<IAzureStorageService>();
+            _userService = A.Fake<IUserService>();
+
+            _fileShareServiceCache = new FileShareServiceCache(_azureTableStorageClient, _cacheConfiguration, _azureStorageService, _logger, _userService);
+        }
+
+        [Test]
+        public async Task WhenFSSCacheCallsGetAllYearWeekFromCache_ThenReturnsNullObject()
+        {
+            A.CallTo(() => _azureStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Returns("testConnectionString");
+
+            A.CallTo(() => _azureTableStorageClient.GetEntityAsync(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored))
+                           .Returns(new CustomTableEntity());
+
+            BatchAttributesSearchModel result = await _fileShareServiceCache.GetAllYearWeekFromCache(string.Empty, string.Empty);
+
+            Assert.IsInstanceOf<BatchAttributesSearchModel>(result);
+            Assert.IsNull(result.Data);
+        }
+
+        [Test]
+        public async Task WhenExceptionInFSSCacheCallsGetAllYearWeekFromCache_ThenReturnsNullObject()
+        {
+            A.CallTo(() => _azureStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Returns("testConnectionString");
+
+            A.CallTo(() => _azureTableStorageClient.GetEntityAsync(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored))
+                           .Throws(new Exception());
+
+            BatchAttributesSearchModel result = await _fileShareServiceCache.GetAllYearWeekFromCache(string.Empty, string.Empty);
+
+            Assert.IsInstanceOf<BatchAttributesSearchModel>(result);
+            Assert.IsNull(result.Data);
+        }
+
+        [Test]
+        public async Task WhenFSSCacheCallsReceivedExpiredCacheData_ThenReturnsEmptyBatchAttributesResponse()
```

Comment:
Name doesn't make sense to me - `WhenFSSCacheCallsReceivedExpiredCacheData`.

Should it be `WhenFSSCacheReceivesExpiredCacheData`?  Or something else?

Reply:
Yes right ..., done required changes